// @fluentui/react-tabster/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    KEYBORG_FOCUSIN: function() {
        return _keyborg.KEYBORG_FOCUSIN;
    },
    TabsterTypes: function() {
        return _tabster.Types;
    },
    applyFocusVisiblePolyfill: function() {
        return _index1.applyFocusVisiblePolyfill;
    },
    createCustomFocusIndicatorStyle: function() {
        return _index1.createCustomFocusIndicatorStyle;
    },
    createFocusOutlineStyle: function() {
        return _index1.createFocusOutlineStyle;
    },
    dispatchGroupperMoveFocusEvent: function() {
        return _tabster.dispatchGroupperMoveFocusEvent;
    },
    dispatchMoverMoveFocusEvent: function() {
        return _tabster.dispatchMoverMoveFocusEvent;
    },
    useArrowNavigationGroup: function() {
        return _index.useArrowNavigationGroup;
    },
    useFocusFinders: function() {
        return _index.useFocusFinders;
    },
    useFocusObserved: function() {
        return _index.useFocusObserved;
    },
    useFocusVisible: function() {
        return _index.useFocusVisible;
    },
    useFocusWithin: function() {
        return _index.useFocusWithin;
    },
    useFocusableGroup: function() {
        return _index.useFocusableGroup;
    },
    useKeyboardNavAttribute: function() {
        return _index.useKeyboardNavAttribute;
    },
    useMergedTabsterAttributes_unstable: function() {
        return _index.useMergedTabsterAttributes_unstable;
    },
    useModalAttributes: function() {
        return _index.useModalAttributes;
    },
    useObservedElement: function() {
        return _index.useObservedElement;
    },
    useOnKeyboardNavigationChange: function() {
        return _index.useOnKeyboardNavigationChange;
    },
    useRestoreFocusSource: function() {
        return _index.useRestoreFocusSource;
    },
    useRestoreFocusTarget: function() {
        return _index.useRestoreFocusTarget;
    },
    useSetKeyboardNavigation: function() {
        return _index.useSetKeyboardNavigation;
    },
    useTabsterAttributes: function() {
        return _index.useTabsterAttributes;
    },
    useUncontrolledFocus: function() {
        return _index.useUncontrolledFocus;
    }
});
const _index = require("./hooks/index");
const _index1 = require("./focus/index");
const _tabster = require("tabster");
const _keyborg = require("keyborg");
