// @fluentui/react-calendar-compat/lib-commonjs/components/CalendarDay/useCalendarDayStyles.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    calendarDayClassNames: function() {
        return calendarDayClassNames;
    },
    useCalendarDayStyles_unstable: function() {
        return useCalendarDayStyles_unstable;
    }
});
const _react = require("@griffel/react");
const calendarDayClassNames = {
    root: 'fui-CalendarDay',
    header: 'fui-CalendarDay__header',
    monthAndYear: 'fui-CalendarDay__monthAndYear',
    monthComponents: 'fui-CalendarDay__monthComponents',
    headerIconButton: 'fui-CalendarDay__headerIconButton',
    disabledStyle: 'fui-CalendarDay__disabledStyle'
};
const useRootStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        B7ck84d: "f1e4lqlz",
        z8tnut: "f1kcqot9",
        z189sj: [
            "f11qrl6u",
            "fjlbh76"
        ],
        Byoj8tv: "fpe6lb7",
        uwmqm3: [
            "fjlbh76",
            "f11qrl6u"
        ],
        a9b677: "f4m2ahc"
    },
    normalize: {
        E5pizo: "f1couhl3",
        B7ck84d: "f1ewtqcl",
        B6of3ja: "f1hu3pq6",
        t21cq0: [
            "f11qmguv",
            "f1tyq0we"
        ],
        jrapky: "f19f4twv",
        Frg6f3: [
            "f1tyq0we",
            "f11qmguv"
        ],
        z8tnut: "f1g0x7ka",
        z189sj: [
            "fhxju0i",
            "f1cnd47f"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "f1cnd47f",
            "fhxju0i"
        ]
    },
    showWeekNumbers: {
        a9b677: "fl524qk"
    }
}, {
    d: [
        ".f1e4lqlz{box-sizing:content-box;}",
        ".f1kcqot9{padding-top:12px;}",
        ".f11qrl6u{padding-right:12px;}",
        ".fjlbh76{padding-left:12px;}",
        ".fpe6lb7{padding-bottom:12px;}",
        ".f4m2ahc{width:196px;}",
        ".f1couhl3{box-shadow:none;}",
        ".f1ewtqcl{box-sizing:border-box;}",
        ".f1hu3pq6{margin-top:0;}",
        ".f11qmguv{margin-right:0;}",
        ".f1tyq0we{margin-left:0;}",
        ".f19f4twv{margin-bottom:0;}",
        ".f1g0x7ka{padding-top:0;}",
        ".fhxju0i{padding-right:0;}",
        ".f1cnd47f{padding-left:0;}",
        ".f1qch9an{padding-bottom:0;}",
        ".fl524qk{width:226px;}"
    ]
});
const useHeaderStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        mc9l5x: "ftuwxu6",
        Bqenvij: "fxldao9",
        Bg96gwp: "f1vzymgo",
        qhf8xq: "f10pi13n",
        a9b677: "fly5x3f"
    }
}, {
    d: [
        ".ftuwxu6{display:inline-flex;}",
        ".fxldao9{height:28px;}",
        ".f1vzymgo{line-height:44px;}",
        ".f10pi13n{position:relative;}",
        ".fly5x3f{width:100%;}"
    ]
});
const useMonthAndYearStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        Bt984gj: "f122n59",
        De3pzq: "f1c21dwh",
        icvyot: "f1ern45e",
        vrafjx: [
            "f1n71otn",
            "f1deefiw"
        ],
        oivjwe: "f1h8hb77",
        wvpqe5: [
            "f1deefiw",
            "f1n71otn"
        ],
        Bbmb7ep: [
            "f2i1086",
            "f1rlsnju"
        ],
        Beyfa6y: [
            "f1rlsnju",
            "f2i1086"
        ],
        B7oj6ja: [
            "f1stktl2",
            "f1127022"
        ],
        Btl43ni: [
            "f1127022",
            "f1stktl2"
        ],
        sj55zd: "f19n0e5",
        mc9l5x: "f14t3ns0",
        Bh6795r: "fqerorx",
        Bahqtrf: "f1mo0ibp",
        Be2twd7: "fkhj508",
        Bhrd7zp: "fl43uef",
        Bg96gwp: "f336tjw",
        B68tc82: "f1p9o1ba",
        Bmxbyg5: "f1sil6mw",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "f8wuabp",
            "fycuoez"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "frdkuqy",
            "f81rol6"
        ],
        fsow6f: [
            "f1o700av",
            "fes3tcz"
        ],
        ygn44y: "f1cmbuwj",
        Huce71: "fz5stix"
    },
    animation: {
        vin17d: "fhsqnsn",
        Bf5fcs: "fywypte",
        Bv12yb3: "f1h7u52p",
        Ezkn3b: "fjq4c94"
    },
    headerIsClickable: {
        Jwef8y: "f121v1wq",
        Bi91k9c: "feu1g3u",
        eoavqd: "f8491dx",
        iro3zm: "f6r9z65"
    }
}, {
    d: [
        ".f122n59{align-items:center;}",
        ".f1c21dwh{background-color:var(--colorTransparentBackground);}",
        ".f1ern45e{border-top-style:none;}",
        ".f1n71otn{border-right-style:none;}",
        ".f1deefiw{border-left-style:none;}",
        ".f1h8hb77{border-bottom-style:none;}",
        ".f2i1086{border-bottom-right-radius:2px;}",
        ".f1rlsnju{border-bottom-left-radius:2px;}",
        ".f1stktl2{border-top-right-radius:2px;}",
        ".f1127022{border-top-left-radius:2px;}",
        ".f19n0e5{color:var(--colorNeutralForeground1);}",
        ".f14t3ns0{display:inline-block;}",
        ".fqerorx{flex-grow:1;}",
        ".f1mo0ibp{font-family:inherit;}",
        ".fkhj508{font-size:var(--fontSizeBase300);}",
        ".fl43uef{font-weight:var(--fontWeightSemibold);}",
        ".f336tjw{line-height:28px;}",
        ".f1p9o1ba{overflow-x:hidden;}",
        ".f1sil6mw{overflow-y:hidden;}",
        ".f1g0x7ka{padding-top:0;}",
        ".f8wuabp{padding-right:4px;}",
        ".fycuoez{padding-left:4px;}",
        ".f1qch9an{padding-bottom:0;}",
        ".frdkuqy{padding-left:10px;}",
        ".f81rol6{padding-right:10px;}",
        ".f1o700av{text-align:left;}",
        ".fes3tcz{text-align:right;}",
        ".f1cmbuwj{text-overflow:ellipsis;}",
        ".fz5stix{white-space:nowrap;}",
        ".fhsqnsn{animation-duration:0.267s;}",
        ".fywypte{animation-fill-mode:both;}",
        ".f1h7u52p{animation-name:f5j8bii;}",
        ".fjq4c94{animation-timing-function:cubic-bezier(.1,.25,.75,.9);}"
    ],
    k: [
        "@keyframes f5j8bii{from{opacity:0;}to{opacity:1;}}"
    ],
    h: [
        ".f121v1wq:hover{background-color:var(--colorBrandBackground2);}",
        ".feu1g3u:hover{color:var(--colorNeutralForeground1Hover);}",
        ".f8491dx:hover{cursor:pointer;}",
        ".f6r9z65:hover:active{background-color:var(--colorBrandBackground2);}"
    ]
});
const useMonthComponentsStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        qb2dma: "fpydfdc",
        mc9l5x: "ftuwxu6"
    }
}, {
    d: [
        ".fpydfdc{align-self:flex-end;}",
        ".ftuwxu6{display:inline-flex;}"
    ]
});
const useHeaderIconButtonStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        De3pzq: "f1c21dwh",
        icvyot: "f1ern45e",
        vrafjx: [
            "f1n71otn",
            "f1deefiw"
        ],
        oivjwe: "f1h8hb77",
        wvpqe5: [
            "f1deefiw",
            "f1n71otn"
        ],
        Bbmb7ep: [
            "f2i1086",
            "f1rlsnju"
        ],
        Beyfa6y: [
            "f1rlsnju",
            "f2i1086"
        ],
        B7oj6ja: [
            "f1stktl2",
            "f1127022"
        ],
        Btl43ni: [
            "f1127022",
            "f1stktl2"
        ],
        sj55zd: "f11d4kpn",
        mc9l5x: "ftgm304",
        Bahqtrf: "f1mo0ibp",
        Be2twd7: "fy9rknc",
        Bqenvij: "fxldao9",
        Bg96gwp: "f336tjw",
        B68tc82: "f1mtd64y",
        Bmxbyg5: "f1y7q3j9",
        z8tnut: "f1g0x7ka",
        z189sj: [
            "fhxju0i",
            "f1cnd47f"
        ],
        Byoj8tv: "f1qch9an",
        uwmqm3: [
            "f1cnd47f",
            "fhxju0i"
        ],
        qhf8xq: "f10pi13n",
        fsow6f: "f17mccla",
        a9b677: "f1w9dchk",
        Jwef8y: "f18a0fwf",
        Bi91k9c: "f3sdaeg",
        eoavqd: "f8491dx",
        Dwtyhm: "f14gx60i",
        ukgb2i: "f1o3croz",
        pugfm4: "f11jyo21",
        iro3zm: "f1wvkgvo",
        B2d53fq: "f1wdoaqd"
    }
}, {
    d: [
        ".f1c21dwh{background-color:var(--colorTransparentBackground);}",
        ".f1ern45e{border-top-style:none;}",
        ".f1n71otn{border-right-style:none;}",
        ".f1deefiw{border-left-style:none;}",
        ".f1h8hb77{border-bottom-style:none;}",
        ".f2i1086{border-bottom-right-radius:2px;}",
        ".f1rlsnju{border-bottom-left-radius:2px;}",
        ".f1stktl2{border-top-right-radius:2px;}",
        ".f1127022{border-top-left-radius:2px;}",
        ".f11d4kpn{color:var(--colorNeutralForeground3);}",
        ".ftgm304{display:block;}",
        ".f1mo0ibp{font-family:inherit;}",
        ".fy9rknc{font-size:var(--fontSizeBase200);}",
        ".fxldao9{height:28px;}",
        ".f336tjw{line-height:28px;}",
        ".f1mtd64y{overflow-x:visible;}",
        ".f1y7q3j9{overflow-y:visible;}",
        ".f1g0x7ka{padding-top:0;}",
        ".fhxju0i{padding-right:0;}",
        ".f1cnd47f{padding-left:0;}",
        ".f1qch9an{padding-bottom:0;}",
        ".f10pi13n{position:relative;}",
        ".f17mccla{text-align:center;}",
        ".f1w9dchk{width:28px;}"
    ],
    h: [
        ".f18a0fwf:hover{background-color:var(--colorBrandBackgroundInvertedHover);}",
        ".f3sdaeg:hover{color:var(--colorBrandForegroundOnLightHover);}",
        ".f8491dx:hover{cursor:pointer;}",
        ".f14gx60i:hover{outline-width:1px;}",
        ".f1o3croz:hover{outline-style:solid;}",
        ".f11jyo21:hover{outline-color:var(--colorTransparentStroke);}",
        ".f1wvkgvo:hover:active{background-color:var(--colorBrandBackgroundInvertedPressed);}",
        ".f1wdoaqd:hover:active{color:var(--colorBrandForegroundOnLightPressed);}"
    ]
});
const useDisabledStyleStyles = /*#__PURE__*/ (0, _react.__styles)({
    base: {
        Bg33ivm: "f1oelgmz",
        t69wg0: "fhh1ekj"
    }
}, {
    d: [
        ".f1oelgmz,.f1oelgmz:disabled,.f1oelgmz button{color:var(--colorNeutralForegroundDisabled);}",
        ".fhh1ekj,.fhh1ekj:disabled,.fhh1ekj button{pointer-events:none;}"
    ]
});
const useCalendarDayStyles_unstable = (props)=>{
    const rootStyles = useRootStyles();
    const headerStyles = useHeaderStyles();
    const monthAndYearStyles = useMonthAndYearStyles();
    const monthComponentsStyles = useMonthComponentsStyles();
    const headerIconButtonStyles = useHeaderIconButtonStyles();
    const disabledStyleStyles = useDisabledStyleStyles();
    const { className, headerIsClickable, showWeekNumbers } = props;
    return {
        root: (0, _react.mergeClasses)(calendarDayClassNames.root, rootStyles.normalize, rootStyles.base, showWeekNumbers && rootStyles.showWeekNumbers, className),
        header: (0, _react.mergeClasses)(calendarDayClassNames.header, headerStyles.base),
        monthAndYear: (0, _react.mergeClasses)(calendarDayClassNames.monthAndYear, monthAndYearStyles.base, monthAndYearStyles.animation, headerIsClickable && monthAndYearStyles.headerIsClickable),
        monthComponents: (0, _react.mergeClasses)(calendarDayClassNames.monthComponents, monthComponentsStyles.base),
        headerIconButton: (0, _react.mergeClasses)(calendarDayClassNames.headerIconButton, headerIconButtonStyles.base),
        disabledStyle: (0, _react.mergeClasses)(calendarDayClassNames.disabledStyle, disabledStyleStyles.base)
    };
}; //# sourceMappingURL=useCalendarDayStyles.styles.js.map
